lib/screens/login/bloc/login_bloc.dart

```diff
@@ -23,7 +22,7 @@ class LoginBloc extends Bloc<LoginEvent, LoginState> {
         final token = await authRepository.login(event.login);
         yield LoginSuccess();
         authBloc.add(JustLoggedIn(token.token));
-      } on Failure catch (failure) {
+      } catch (failure) {
         yield LoginFailure(failure.message);
```

Comment:
I think I created `Failure` class to represent errors that are expected and can happen.

If you do something like this, we'll catch _all_ error, even the ones we don't expect to happen (e.g I can think of some OutOfMemory error or its dart equivalent).

Reply:
Not sure why i did this, iirc was facing some error in login bloc test. But isn't catching error good? If we don't get a error for example a null pointer exception,  we would not be able to find out tbe cause. Is that making sense? Not sure why you made the Failure class

Reply:
is this comment solved? @Techno-Disaster

Reply:
imo yes. @bartekpacia what do you think?

Reply:
also @isabelcosta if we need it we can just add it later. It's not a breaking change at all.

Reply:
I thought of something like this:

```
on Failure catch (failure) {
      } catch (failure) {
        yield LoginFailure(failure.message);
on dynamic catch (anything) {
        // handle e.g NPEs
}
```

Reply:
`Failure` is for stuff that we know can happen (it's thrown by Repositories when e.g `http 500` occurs, which of course can happen.
But when something bizarre happens, `Repository`will return an exception and we need to catch it using a typical `catch (everything)`

Reply:
@Techno-Disaster if there is anything here that can be turned into a new issue please create it. I am okay with approving this if we take notes of what needs to be done next.

Reply:
@isabelcosta If any bug comes up due to this I'll create an issue asap. But for now there are no breaking changes as such. I have pinged Bartek below for help on why I removed ```Faliure``` in thr first place